ndk/src/hardware_buffer_format.rs

```diff
@@ -35,4 +36,43 @@ pub enum HardwareBufferFormat {
     S8_UINT = ffi::AHardwareBuffer_Format::AHARDWAREBUFFER_FORMAT_S8_UINT.0,
     #[cfg(feature = "api-level-26")]
     Y8Cb8Cr8_420 = ffi::AHardwareBuffer_Format::AHARDWAREBUFFER_FORMAT_Y8Cb8Cr8_420.0,
+    #[cfg(feature = "api-level-26")]
+    YCbCr_P010 = ffi::AHardwareBuffer_Format::AHARDWAREBUFFER_FORMAT_YCbCr_P010.0,
+    #[cfg(feature = "api-level-26")]
+    R8_UNORM = ffi::AHardwareBuffer_Format::AHARDWAREBUFFER_FORMAT_R8_UNORM.0,
+}
+
+impl HardwareBufferFormat {
+    pub fn bytes_per_pixel(self) -> usize {
+        match self {
+            Self::R8G8B8A8_UNORM | Self::R8G8B8X8_UNORM => 4,
+            #[cfg(feature = "api-level-26")]
+            Self::R8G8B8_UNORM => 3,
+            Self::R5G6B5_UNORM => 2,
+            #[cfg(feature = "api-level-26")]
+            Self::R16G16B16A16_FLOAT => 8,
+            #[cfg(feature = "api-level-26")]
+            Self::R10G10B10A2_UNORM => 4,
+            #[cfg(feature = "api-level-26")]
+            Self::BLOB => 1,
+            #[cfg(feature = "api-level-26")]
+            Self::D16_UNORM => 2,
+            #[cfg(feature = "api-level-26")]
+            Self::D24_UNORM => 3,
+            #[cfg(feature = "api-level-26")]
+            Self::D24_UNORM_S8_UINT => 4,
+            #[cfg(feature = "api-level-26")]
+            Self::D32_FLOAT => 4,
+            #[cfg(feature = "api-level-26")]
+            Self::D32_FLOAT_S8_UINT => 5,
+            #[cfg(feature = "api-level-26")]
+            Self::S8_UINT => 1,
+            #[cfg(feature = "api-level-26")]
+            Self::Y8Cb8Cr8_420 => 3,
+            #[cfg(feature = "api-level-26")]
+            Self::YCbCr_P010 => unimplemented!("Semiplanar"),
```

Comment:
ah, right, I was wondering if there were going to be planar video formats to handle

panicking for planar video formats is a bit harsh - maybe this should return an Option though because there's no practical bytes per pixel for video formats with different sized planes.

In practice as long as we cover the common rgbx formats that should be all that's required for helping enable apps to render from the CPU, whereas video formats will show up if your pulling frames from a camera and more context-aware code won't be trying to ask for a bytes_per_pixel for those formats anyway.

Reply:
Returning an option now and documented this.